src/Glpi/Exception/Http/AuthenticationFailedHttpException.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Exception\Http;
+
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class AuthenticationFailedHttpException extends HttpException
+{
+    use HttpExceptionTrait;
+
+    public function __construct(string $message = '', ?\Throwable $previous = null, int $code = 0, array $headers = [])
+    {
+        parent::__construct(401, $message, $previous, $headers, $code);
```

Comment:
According to [the RFC](https://datatracker.ietf.org/doc/html/rfc7235#section-3.1):

> The server generating a 401 response MUST send a WWW-Authenticate header field ([Section 4.1](https://datatracker.ietf.org/doc/html/rfc7235#section-4.1)) containing at least one challenge applicable to the target resource.

The constructor of `\Symfony\Component\HttpKernel\Exception\UnauthorizedHttpException` already has this, and it should be covered in GLPI's `ErrorController`

Reply:
In fact, as this exception is caught by a listener and never sent to the error controller, it was not necessary to extend the `Symfony\Component\HttpKernel\Exception\HttpException` exception.

I changed it in the `Session/Auth errors are not real http exceptions` commit.